File: SampleCode/RestApi/Clients/Net.RestSharp/EmbyClient.Dotnet.Beta/Api/GenericUIApiServiceApi.cs
/*
 * EmbyClient.Dotnet.Beta
 */

using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using RestSharp;
using EmbyClient.Dotnet.Beta.Client;
using EmbyClient.Dotnet.Beta.Model;

namespace EmbyClient.Dotnet.Beta.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
        public interface IGenericUIApiServiceApi : IApiAccessor
    {
        #region Synchronous Operations
        /// <summary>
        /// Gets UI view data
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>EmbyWebGenericUIModelUIViewInfo</returns>
        EmbyWebGenericUIModelUIViewInfo GetUIView (string pageId, string clientLocale);

        /// <summary>
        /// Gets UI view data
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>ApiResponse of EmbyWebGenericUIModelUIViewInfo</returns>
        ApiResponse<EmbyWebGenericUIModelUIViewInfo> GetUIViewWithHttpInfo (string pageId, string clientLocale);
        /// <summary>
        /// Execute a command in the context of tv setup
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>EmbyWebGenericUIModelUIViewInfo</returns>
        EmbyWebGenericUIModelUIViewInfo PostUICommand (EmbyWebGenericUIApiEndpointsRunUICommand body);

        /// <summary>
        /// Execute a command in the context of tv setup
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>ApiResponse of EmbyWebGenericUIModelUIViewInfo</returns>
        ApiResponse<EmbyWebGenericUIModelUIViewInfo> PostUICommandWithHttpInfo (EmbyWebGenericUIApiEndpointsRunUICommand body);
        #endregion Synchronous Operations
        #region Asynchronous Operations
        /// <summary>
        /// Gets UI view data
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>Task of EmbyWebGenericUIModelUIViewInfo</returns>
        System.Threading.Tasks.Task<EmbyWebGenericUIModelUIViewInfo> GetUIViewAsync (string pageId, string clientLocale);

        /// <summary>
        /// Gets UI view data
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>Task of ApiResponse (EmbyWebGenericUIModelUIViewInfo)</returns>
        System.Threading.Tasks.Task<ApiResponse<EmbyWebGenericUIModelUIViewInfo>> GetUIViewAsyncWithHttpInfo (string pageId, string clientLocale);
        /// <summary>
        /// Execute a command in the context of tv setup
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>Task of EmbyWebGenericUIModelUIViewInfo</returns>
        System.Threading.Tasks.Task<EmbyWebGenericUIModelUIViewInfo> PostUICommandAsync (EmbyWebGenericUIApiEndpointsRunUICommand body);

        /// <summary>
        /// Execute a command in the context of tv setup
        /// </summary>
        /// <remarks>
        /// Requires authentication as user
        /// </remarks>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>Task of ApiResponse (EmbyWebGenericUIModelUIViewInfo)</returns>
        System.Threading.Tasks.Task<ApiResponse<EmbyWebGenericUIModelUIViewInfo>> PostUICommandAsyncWithHttpInfo (EmbyWebGenericUIApiEndpointsRunUICommand body);
        #endregion Asynchronous Operations
    }

    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
        public partial class GenericUIApiServiceApi : IGenericUIApiServiceApi
    {
        private EmbyClient.Dotnet.Beta.Client.ExceptionFactory _exceptionFactory = (name, response) => null;

        /// <summary>
        /// Initializes a new instance of the <see cref="GenericUIApiServiceApi"/> class.
        /// </summary>
        /// <returns></returns>
        public GenericUIApiServiceApi(String basePath)
        {
            this.Configuration = new EmbyClient.Dotnet.Beta.Client.Configuration { BasePath = basePath };

            ExceptionFactory = EmbyClient.Dotnet.Beta.Client.Configuration.DefaultExceptionFactory;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="GenericUIApiServiceApi"/> class
        /// </summary>
        /// <returns></returns>
        public GenericUIApiServiceApi()
        {
            this.Configuration = EmbyClient.Dotnet.Beta.Client.Configuration.Default;

            ExceptionFactory = EmbyClient.Dotnet.Beta.Client.Configuration.DefaultExceptionFactory;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="GenericUIApiServiceApi"/> class
        /// using Configuration object
        /// </summary>
        /// <param name="configuration">An instance of Configuration</param>
        /// <returns></returns>
        public GenericUIApiServiceApi(EmbyClient.Dotnet.Beta.Client.Configuration configuration = null)
        {
            if (configuration == null) // use the default one in Configuration
                this.Configuration = EmbyClient.Dotnet.Beta.Client.Configuration.Default;
            else
                this.Configuration = configuration;

            ExceptionFactory = EmbyClient.Dotnet.Beta.Client.Configuration.DefaultExceptionFactory;
        }

        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <value>The base path</value>
        public String GetBasePath()
        {
            return this.Configuration.ApiClient.RestClient.BaseUrl.ToString();
        }

        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <value>The base path</value>
        [Obsolete("SetBasePath is deprecated, please do 'Configuration.ApiClient = new ApiClient(\"http://new-path\")' instead.")]
        public void SetBasePath(String basePath)
        {
            // do nothing
        }

        /// <summary>
        /// Gets or sets the configuration object
        /// </summary>
        /// <value>An instance of the Configuration</value>
        public EmbyClient.Dotnet.Beta.Client.Configuration Configuration {get; set;}

        /// <summary>
        /// Provides a factory method hook for the creation of exceptions.
        /// </summary>
        public EmbyClient.Dotnet.Beta.Client.ExceptionFactory ExceptionFactory
        {
            get
            {
                if (_exceptionFactory != null && _exceptionFactory.GetInvocationList().Length > 1)
                {
                    throw new InvalidOperationException("Multicast delegate for ExceptionFactory is unsupported.");
                }
                return _exceptionFactory;
            }
            set { _exceptionFactory = value; }
        }

        /// <summary>
        /// Gets the default header.
        /// </summary>
        /// <returns>Dictionary of HTTP header</returns>
        [Obsolete("DefaultHeader is deprecated, please use Configuration.DefaultHeader instead.")]
        public IDictionary<String, String> DefaultHeader()
        {
            return new ReadOnlyDictionary<string, string>(this.Configuration.DefaultHeader);
        }

        /// <summary>
        /// Add default header.
        /// </summary>
        /// <param name="key">Header field name.</param>
        /// <param name="value">Header field value.</param>
        /// <returns></returns>
        [Obsolete("AddDefaultHeader is deprecated, please use Configuration.AddDefaultHeader instead.")]
        public void AddDefaultHeader(string key, string value)
        {
            this.Configuration.AddDefaultHeader(key, value);
        }

        /// <summary>
        /// Gets UI view data Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>EmbyWebGenericUIModelUIViewInfo</returns>
        public EmbyWebGenericUIModelUIViewInfo GetUIView (string pageId, string clientLocale)
        {
             ApiResponse<EmbyWebGenericUIModelUIViewInfo> localVarResponse = GetUIViewWithHttpInfo(pageId, clientLocale);
             return localVarResponse.Data;
        }

        /// <summary>
        /// Gets UI view data Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>ApiResponse of EmbyWebGenericUIModelUIViewInfo</returns>
        public ApiResponse< EmbyWebGenericUIModelUIViewInfo > GetUIViewWithHttpInfo (string pageId, string clientLocale)
        {
            // verify the required parameter 'pageId' is set
            if (pageId == null)
                throw new ApiException(400, "Missing required parameter 'pageId' when calling GenericUIApiServiceApi->GetUIView");
            // verify the required parameter 'clientLocale' is set
            if (clientLocale == null)
                throw new ApiException(400, "Missing required parameter 'clientLocale' when calling GenericUIApiServiceApi->GetUIView");

            var localVarPath = "/UI/View";
            var localVarPathParams = new Dictionary<String, String>();
            var localVarQueryParams = new List<KeyValuePair<String, String>>();
            var localVarHeaderParams = new Dictionary<String, String>(this.Configuration.DefaultHeader);
            var localVarFormParams = new Dictionary<String, String>();
            var localVarFileParams = new Dictionary<String, FileParameter>();
            Object localVarPostBody = null;

            // to determine the Content-Type header
            String[] localVarHttpContentTypes = new String[] {
            };
            String localVarHttpContentType = this.Configuration.ApiClient.SelectHeaderContentType(localVarHttpContentTypes);

            // to determine the Accept header
            String[] localVarHttpHeaderAccepts = new String[] {
                "application/json",
                "application/xml"
            };
            String localVarHttpHeaderAccept = this.Configuration.ApiClient.SelectHeaderAccept(localVarHttpHeaderAccepts);
            if (localVarHttpHeaderAccept != null)
                localVarHeaderParams.Add("Accept", localVarHttpHeaderAccept);

            if (pageId != null) localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "PageId", pageId)); // query parameter
            if (clientLocale != null) localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "ClientLocale", clientLocale)); // query parameter
            // authentication (apikeyauth) required
            if (!String.IsNullOrEmpty(this.Configuration.GetApiKeyWithPrefix("api_key")))
            {
                localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "api_key", this.Configuration.GetApiKeyWithPrefix("api_key")));
            }
            // authentication (embyauth) required
            // bearer required
            if (!String.IsNullOrEmpty(this.Configuration.AccessToken))
            {
                localVarHeaderParams["Authorization"] = "Bearer " + this.Configuration.AccessToken;
            }

            // make the HTTP request
            IRestResponse localVarResponse = (IRestResponse) this.Configuration.ApiClient.CallApi(localVarPath,
                Method.GET, localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarFileParams,
                localVarPathParams, localVarHttpContentType);

            int localVarStatusCode = (int) localVarResponse.StatusCode;

            if (ExceptionFactory != null)
            {
                Exception exception = ExceptionFactory("GetUIView", localVarResponse);
                if (exception != null) throw exception;
            }

            return new ApiResponse<EmbyWebGenericUIModelUIViewInfo>(localVarStatusCode,
                localVarResponse.Headers.ToDictionary(x => x.Name, x => string.Join(",", x.Value)),
                (EmbyWebGenericUIModelUIViewInfo) this.Configuration.ApiClient.Deserialize(localVarResponse, typeof(EmbyWebGenericUIModelUIViewInfo)));
        }

        /// <summary>
        /// Gets UI view data Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>Task of EmbyWebGenericUIModelUIViewInfo</returns>
        public async System.Threading.Tasks.Task<EmbyWebGenericUIModelUIViewInfo> GetUIViewAsync (string pageId, string clientLocale)
        {
             ApiResponse<EmbyWebGenericUIModelUIViewInfo> localVarResponse = await GetUIViewAsyncWithHttpInfo(pageId, clientLocale);
             return localVarResponse.Data;

        }

        /// <summary>
        /// Gets UI view data Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="pageId">Id of the page controller</param>
        /// <param name="clientLocale">Locale identifier of the client</param>
        /// <returns>Task of ApiResponse (EmbyWebGenericUIModelUIViewInfo)</returns>
        public async System.Threading.Tasks.Task<ApiResponse<EmbyWebGenericUIModelUIViewInfo>> GetUIViewAsyncWithHttpInfo (string pageId, string clientLocale)
        {
            // verify the required parameter 'pageId' is set
            if (pageId == null)
                throw new ApiException(400, "Missing required parameter 'pageId' when calling GenericUIApiServiceApi->GetUIView");
            // verify the required parameter 'clientLocale' is set
            if (clientLocale == null)
                throw new ApiException(400, "Missing required parameter 'clientLocale' when calling GenericUIApiServiceApi->GetUIView");

            var localVarPath = "/UI/View";
            var localVarPathParams = new Dictionary<String, String>();
            var localVarQueryParams = new List<KeyValuePair<String, String>>();
            var localVarHeaderParams = new Dictionary<String, String>(this.Configuration.DefaultHeader);
            var localVarFormParams = new Dictionary<String, String>();
            var localVarFileParams = new Dictionary<String, FileParameter>();
            Object localVarPostBody = null;

            // to determine the Content-Type header
            String[] localVarHttpContentTypes = new String[] {
            };
            String localVarHttpContentType = this.Configuration.ApiClient.SelectHeaderContentType(localVarHttpContentTypes);

            // to determine the Accept header
            String[] localVarHttpHeaderAccepts = new String[] {
                "application/json",
                "application/xml"
            };
            String localVarHttpHeaderAccept = this.Configuration.ApiClient.SelectHeaderAccept(localVarHttpHeaderAccepts);
            if (localVarHttpHeaderAccept != null)
                localVarHeaderParams.Add("Accept", localVarHttpHeaderAccept);

            if (pageId != null) localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "PageId", pageId)); // query parameter
            if (clientLocale != null) localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "ClientLocale", clientLocale)); // query parameter
            // authentication (apikeyauth) required
            if (!String.IsNullOrEmpty(this.Configuration.GetApiKeyWithPrefix("api_key")))
            {
                localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "api_key", this.Configuration.GetApiKeyWithPrefix("api_key")));
            }
            // authentication (embyauth) required
            // bearer required
            if (!String.IsNullOrEmpty(this.Configuration.AccessToken))
            {
                localVarHeaderParams["Authorization"] = "Bearer " + this.Configuration.AccessToken;
            }

            // make the HTTP request
            IRestResponse localVarResponse = (IRestResponse) await this.Configuration.ApiClient.CallApiAsync(localVarPath,
                Method.GET, localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarFileParams,
                localVarPathParams, localVarHttpContentType);

            int localVarStatusCode = (int) localVarResponse.StatusCode;

            if (ExceptionFactory != null)
            {
                Exception exception = ExceptionFactory("GetUIView", localVarResponse);
                if (exception != null) throw exception;
            }

            return new ApiResponse<EmbyWebGenericUIModelUIViewInfo>(localVarStatusCode,
                localVarResponse.Headers.ToDictionary(x => x.Name, x => string.Join(",", x.Value)),
                (EmbyWebGenericUIModelUIViewInfo) this.Configuration.ApiClient.Deserialize(localVarResponse, typeof(EmbyWebGenericUIModelUIViewInfo)));
        }

        /// <summary>
        /// Execute a command in the context of tv setup Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>EmbyWebGenericUIModelUIViewInfo</returns>
        public EmbyWebGenericUIModelUIViewInfo PostUICommand (EmbyWebGenericUIApiEndpointsRunUICommand body)
        {
             ApiResponse<EmbyWebGenericUIModelUIViewInfo> localVarResponse = PostUICommandWithHttpInfo(body);
             return localVarResponse.Data;
        }

        /// <summary>
        /// Execute a command in the context of tv setup Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>ApiResponse of EmbyWebGenericUIModelUIViewInfo</returns>
        public ApiResponse< EmbyWebGenericUIModelUIViewInfo > PostUICommandWithHttpInfo (EmbyWebGenericUIApiEndpointsRunUICommand body)
        {
            // verify the required parameter 'body' is set
            if (body == null)
                throw new ApiException(400, "Missing required parameter 'body' when calling GenericUIApiServiceApi->PostUICommand");

            var localVarPath = "/UI/Command";
            var localVarPathParams = new Dictionary<String, String>();
            var localVarQueryParams = new List<KeyValuePair<String, String>>();
            var localVarHeaderParams = new Dictionary<String, String>(this.Configuration.DefaultHeader);
            var localVarFormParams = new Dictionary<String, String>();
            var localVarFileParams = new Dictionary<String, FileParameter>();
            Object localVarPostBody = null;

            // to determine the Content-Type header
            String[] localVarHttpContentTypes = new String[] {
                "application/json", 
                "application/xml"
            };
            String localVarHttpContentType = this.Configuration.ApiClient.SelectHeaderContentType(localVarHttpContentTypes);

            // to determine the Accept header
            String[] localVarHttpHeaderAccepts = new String[] {
                "application/json",
                "application/xml"
            };
            String localVarHttpHeaderAccept = this.Configuration.ApiClient.SelectHeaderAccept(localVarHttpHeaderAccepts);
            if (localVarHttpHeaderAccept != null)
                localVarHeaderParams.Add("Accept", localVarHttpHeaderAccept);

            if (body != null && body.GetType() != typeof(byte[]))
            {
                localVarPostBody = this.Configuration.ApiClient.Serialize(body); // http body (model) parameter
            }
            else
            {
                localVarPostBody = body; // byte array
            }
            // authentication (apikeyauth) required
            if (!String.IsNullOrEmpty(this.Configuration.GetApiKeyWithPrefix("api_key")))
            {
                localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "api_key", this.Configuration.GetApiKeyWithPrefix("api_key")));
            }
            // authentication (embyauth) required
            // bearer required
            if (!String.IsNullOrEmpty(this.Configuration.AccessToken))
            {
                localVarHeaderParams["Authorization"] = "Bearer " + this.Configuration.AccessToken;
            }

            // make the HTTP request
            IRestResponse localVarResponse = (IRestResponse) this.Configuration.ApiClient.CallApi(localVarPath,
                Method.POST, localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarFileParams,
                localVarPathParams, localVarHttpContentType);

            int localVarStatusCode = (int) localVarResponse.StatusCode;

            if (ExceptionFactory != null)
            {
                Exception exception = ExceptionFactory("PostUICommand", localVarResponse);
                if (exception != null) throw exception;
            }

            return new ApiResponse<EmbyWebGenericUIModelUIViewInfo>(localVarStatusCode,
                localVarResponse.Headers.ToDictionary(x => x.Name, x => string.Join(",", x.Value)),
                (EmbyWebGenericUIModelUIViewInfo) this.Configuration.ApiClient.Deserialize(localVarResponse, typeof(EmbyWebGenericUIModelUIViewInfo)));
        }

        /// <summary>
        /// Execute a command in the context of tv setup Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>Task of EmbyWebGenericUIModelUIViewInfo</returns>
        public async System.Threading.Tasks.Task<EmbyWebGenericUIModelUIViewInfo> PostUICommandAsync (EmbyWebGenericUIApiEndpointsRunUICommand body)
        {
             ApiResponse<EmbyWebGenericUIModelUIViewInfo> localVarResponse = await PostUICommandAsyncWithHttpInfo(body);
             return localVarResponse.Data;

        }

        /// <summary>
        /// Execute a command in the context of tv setup Requires authentication as user
        /// </summary>
        /// <exception cref="EmbyClient.Dotnet.Beta.Client.ApiException">Thrown when fails to make API call</exception>
        /// <param name="body">RunUICommand</param>
        /// <returns>Task of ApiResponse (EmbyWebGenericUIModelUIViewInfo)</returns>
        public async System.Threading.Tasks.Task<ApiResponse<EmbyWebGenericUIModelUIViewInfo>> PostUICommandAsyncWithHttpInfo (EmbyWebGenericUIApiEndpointsRunUICommand body)
        {
            // verify the required parameter 'body' is set
            if (body == null)
                throw new ApiException(400, "Missing required parameter 'body' when calling GenericUIApiServiceApi->PostUICommand");

            var localVarPath = "/UI/Command";
            var localVarPathParams = new Dictionary<String, String>();
            var localVarQueryParams = new List<KeyValuePair<String, String>>();
            var localVarHeaderParams = new Dictionary<String, String>(this.Configuration.DefaultHeader);
            var localVarFormParams = new Dictionary<String, String>();
            var localVarFileParams = new Dictionary<String, FileParameter>();
            Object localVarPostBody = null;

            // to determine the Content-Type header
            String[] localVarHttpContentTypes = new String[] {
                "application/json", 
                "application/xml"
            };
            String localVarHttpContentType = this.Configuration.ApiClient.SelectHeaderContentType(localVarHttpContentTypes);

            // to determine the Accept header
            String[] localVarHttpHeaderAccepts = new String[] {
                "application/json",
                "application/xml"
            };
            String localVarHttpHeaderAccept = this.Configuration.ApiClient.SelectHeaderAccept(localVarHttpHeaderAccepts);
            if (localVarHttpHeaderAccept != null)
                localVarHeaderParams.Add("Accept", localVarHttpHeaderAccept);

            if (body != null && body.GetType() != typeof(byte[]))
            {
                localVarPostBody = this.Configuration.ApiClient.Serialize(body); // http body (model) parameter
            }
            else
            {
                localVarPostBody = body; // byte array
            }
            // authentication (apikeyauth) required
            if (!String.IsNullOrEmpty(this.Configuration.GetApiKeyWithPrefix("api_key")))
            {
                localVarQueryParams.AddRange(this.Configuration.ApiClient.ParameterToKeyValuePairs("", "api_key", this.Configuration.GetApiKeyWithPrefix("api_key")));
            }
            // authentication (embyauth) required
            // bearer required
            if (!String.IsNullOrEmpty(this.Configuration.AccessToken))
            {
                localVarHeaderParams["Authorization"] = "Bearer " + this.Configuration.AccessToken;
            }

            // make the HTTP request
            IRestResponse localVarResponse = (IRestResponse) await this.Configuration.ApiClient.CallApiAsync(localVarPath,
                Method.POST, localVarQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarFileParams,
                localVarPathParams, localVarHttpContentType);

            int localVarStatusCode = (int) localVarResponse.StatusCode;

            if (ExceptionFactory != null)
            {
                Exception exception = ExceptionFactory("PostUICommand", localVarResponse);
                if (exception != null) throw exception;
            }

            return new ApiResponse<EmbyWebGenericUIModelUIViewInfo>(localVarStatusCode,
                localVarResponse.Headers.ToDictionary(x => x.Name, x => string.Join(",", x.Value)),
                (EmbyWebGenericUIModelUIViewInfo) this.Configuration.ApiClient.Deserialize(localVarResponse, typeof(EmbyWebGenericUIModelUIViewInfo)));
        }

    }
}

File: SampleCode/RestApi/Clients/Java/src/main/java/EmbyClient/Java/GenericUiApiServiceApi.java
/*
 * Emby Server REST API
 * 
 */

package EmbyClient.Java;

import EmbyClient.ApiCallback;
import EmbyClient.ApiClient;
import EmbyClient.ApiException;
import EmbyClient.ApiResponse;
import EmbyClient.Configuration;
import EmbyClient.Pair;
import EmbyClient.ProgressRequestBody;
import EmbyClient.ProgressResponseBody;

import com.google.gson.reflect.TypeToken;

import java.io.IOException;


import io.swagger.client.model.RunUICommand;
import io.swagger.client.model.UIViewInfo;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class GenericUiApiServiceApi {
    private ApiClient apiClient;

    public GenericUiApiServiceApi() {
        this(Configuration.getDefaultApiClient());
    }

    public GenericUiApiServiceApi(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    public ApiClient getApiClient() {
        return apiClient;
    }

    public void setApiClient(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    /**
     * Build call for getUIView
     * @param pageId Id of the page controller (required)
     * @param clientLocale Locale identifier of the client (required)
     * @param progressListener Progress listener
     * @param progressRequestListener Progress request listener
     * @return Call to execute
     * @throws ApiException If fail to serialize the request body object
     */
    public com.squareup.okhttp.Call getUIViewCall(String pageId, String clientLocale, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        Object localVarPostBody = null;
        
        // create path and map variables
        String localVarPath = "/UI/View";

        List<Pair> localVarQueryParams = new ArrayList<Pair>();
        List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();
        if (pageId != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("PageId", pageId));
        if (clientLocale != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("ClientLocale", clientLocale));

        Map<String, String> localVarHeaderParams = new HashMap<String, String>();

        Map<String, Object> localVarFormParams = new HashMap<String, Object>();

        final String[] localVarAccepts = {
            "application/json", "application/xml"
        };
        final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
        if (localVarAccept != null) localVarHeaderParams.put("Accept", localVarAccept);

        final String[] localVarContentTypes = {
            
        };
        final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
        localVarHeaderParams.put("Content-Type", localVarContentType);

        if(progressListener != null) {
            apiClient.getHttpClient().networkInterceptors().add(new com.squareup.okhttp.Interceptor() {
                @Override
                public com.squareup.okhttp.Response intercept(com.squareup.okhttp.Interceptor.Chain chain) throws IOException {
                    com.squareup.okhttp.Response originalResponse = chain.proceed(chain.request());
                    return originalResponse.newBuilder()
                    .body(new ProgressResponseBody(originalResponse.body(), progressListener))
                    .build();
                }
            });
        }

        String[] localVarAuthNames = new String[] { "apikeyauth", "embyauth" };
        return apiClient.buildCall(localVarPath, "GET", localVarQueryParams, localVarCollectionQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAuthNames, progressRequestListener);
    }
    
    @SuppressWarnings("rawtypes")
    private com.squareup.okhttp.Call getUIViewValidateBeforeCall(String pageId, String clientLocale, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        // verify the required parameter 'pageId' is set
        if (pageId == null) {
            throw new ApiException("Missing the required parameter 'pageId' when calling getUIView(Async)");
        }
        // verify the required parameter 'clientLocale' is set
        if (clientLocale == null) {
            throw new ApiException("Missing the required parameter 'clientLocale' when calling getUIView(Async)");
        }
        
        com.squareup.okhttp.Call call = getUIViewCall(pageId, clientLocale, progressListener, progressRequestListener);
        return call;

        
        
        
        
    }

    /**
     * Gets UI view data
     * Requires authentication as user
     * @param pageId Id of the page controller (required)
     * @param clientLocale Locale identifier of the client (required)
     * @return UIViewInfo
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     */
    public UIViewInfo getUIView(String pageId, String clientLocale) throws ApiException {
        ApiResponse<UIViewInfo> resp = getUIViewWithHttpInfo(pageId, clientLocale);
        return resp.getData();
    }

    /**
     * Gets UI view data
     * Requires authentication as user
     * @param pageId Id of the page controller (required)
     * @param clientLocale Locale identifier of the client (required)
     * @return ApiResponse&lt;UIViewInfo&gt;
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     */
    public ApiResponse<UIViewInfo> getUIViewWithHttpInfo(String pageId, String clientLocale) throws ApiException {
        com.squareup.okhttp.Call call = getUIViewValidateBeforeCall(pageId, clientLocale, null, null);
        Type localVarReturnType = new TypeToken<UIViewInfo>(){}.getType();
        return apiClient.execute(call, localVarReturnType);
    }

    /**
     * Gets UI view data (asynchronously)
     * Requires authentication as user
     * @param pageId Id of the page controller (required)
     * @param clientLocale Locale identifier of the client (required)
     * @param callback The callback to be executed when the API call finishes
     * @return The request call
     * @throws ApiException If fail to process the API call, e.g. serializing the request body object
     */
    public com.squareup.okhttp.Call getUIViewAsync(String pageId, String clientLocale, final ApiCallback<UIViewInfo> callback) throws ApiException {

        ProgressResponseBody.ProgressListener progressListener = null;
        ProgressRequestBody.ProgressRequestListener progressRequestListener = null;

        if (callback != null) {
            progressListener = new ProgressResponseBody.ProgressListener() {
                @Override
                public void update(long bytesRead, long contentLength, boolean done) {
                    callback.onDownloadProgress(bytesRead, contentLength, done);
                }
            };

            progressRequestListener = new ProgressRequestBody.ProgressRequestListener() {
                @Override
                public void onRequestProgress(long bytesWritten, long contentLength, boolean done) {
                    callback.onUploadProgress(bytesWritten, contentLength, done);
                }
            };
        }

        com.squareup.okhttp.Call call = getUIViewValidateBeforeCall(pageId, clientLocale, progressListener, progressRequestListener);
        Type localVarReturnType = new TypeToken<UIViewInfo>(){}.getType();
        apiClient.executeAsync(call, localVarReturnType, callback);
        return call;
    }
    /**
     * Build call for postUICommand
     * @param body RunUICommand (required)
     * @param progressListener Progress listener
     * @param progressRequestListener Progress request listener
     * @return Call to execute
     * @throws ApiException If fail to serialize the request body object
     */
    public com.squareup.okhttp.Call postUICommandCall(RunUICommand body, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        Object localVarPostBody = body;
        
        // create path and map variables
        String localVarPath = "/UI/Command";

        List<Pair> localVarQueryParams = new ArrayList<Pair>();
        List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();

        Map<String, String> localVarHeaderParams = new HashMap<String, String>();

        Map<String, Object> localVarFormParams = new HashMap<String, Object>();

        final String[] localVarAccepts = {
            "application/json", "application/xml"
        };
        final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
        if (localVarAccept != null) localVarHeaderParams.put("Accept", localVarAccept);

        final String[] localVarContentTypes = {
            "application/json", "application/xml"
        };
        final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
        localVarHeaderParams.put("Content-Type", localVarContentType);

        if(progressListener != null) {
            apiClient.getHttpClient().networkInterceptors().add(new com.squareup.okhttp.Interceptor() {
                @Override
                public com.squareup.okhttp.Response intercept(com.squareup.okhttp.Interceptor.Chain chain) throws IOException {
                    com.squareup.okhttp.Response originalResponse = chain.proceed(chain.request());
                    return originalResponse.newBuilder()
                    .body(new ProgressResponseBody(originalResponse.body(), progressListener))
                    .build();
                }
            });
        }

        String[] localVarAuthNames = new String[] { "apikeyauth", "embyauth" };
        return apiClient.buildCall(localVarPath, "POST", localVarQueryParams, localVarCollectionQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAuthNames, progressRequestListener);
    }
    
    @SuppressWarnings("rawtypes")
    private com.squareup.okhttp.Call postUICommandValidateBeforeCall(RunUICommand body, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        // verify the required parameter 'body' is set
        if (body == null) {
            throw new ApiException("Missing the required parameter 'body' when calling postUICommand(Async)");
        }
        
        com.squareup.okhttp.Call call = postUICommandCall(body, progressListener, progressRequestListener);
        return call;

        
        
        
        
    }

    /**
     * Execute a command in the context of tv setup
     * Requires authentication as user
     * @param body RunUICommand (required)
     * @return UIViewInfo
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     */
    public UIViewInfo postUICommand(RunUICommand body) throws ApiException {
        ApiResponse<UIViewInfo> resp = postUICommandWithHttpInfo(body);
        return resp.getData();
    }

    /**
     * Execute a command in the context of tv setup
     * Requires authentication as user
     * @param body RunUICommand (required)
     * @return ApiResponse&lt;UIViewInfo&gt;
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     */
    public ApiResponse<UIViewInfo> postUICommandWithHttpInfo(RunUICommand body) throws ApiException {
        com.squareup.okhttp.Call call = postUICommandValidateBeforeCall(body, null, null);
        Type localVarReturnType = new TypeToken<UIViewInfo>(){}.getType();
        return apiClient.execute(call, localVarReturnType);
    }

    /**
     * Execute a command in the context of tv setup (asynchronously)
     * Requires authentication as user
     * @param body RunUICommand (required)
     * @param callback The callback to be executed when the API call finishes
     * @return The request call
     * @throws ApiException If fail to process the API call, e.g. serializing the request body object
     */
    public com.squareup.okhttp.Call postUICommandAsync(RunUICommand body, final ApiCallback<UIViewInfo> callback) throws ApiException {

        ProgressResponseBody.ProgressListener progressListener = null;
        ProgressRequestBody.ProgressRequestListener progressRequestListener = null;

        if (callback != null) {
            progressListener = new ProgressResponseBody.ProgressListener() {
                @Override
                public void update(long bytesRead, long contentLength, boolean done) {
                    callback.onDownloadProgress(bytesRead, contentLength, done);
                }
            };

            progressRequestListener = new ProgressRequestBody.ProgressRequestListener() {
                @Override
                public void onRequestProgress(long bytesWritten, long contentLength, boolean done) {
                    callback.onUploadProgress(bytesWritten, contentLength, done);
                }
            };
        }

        com.squareup.okhttp.Call call = postUICommandValidateBeforeCall(body, progressListener, progressRequestListener);
        Type localVarReturnType = new TypeToken<UIViewInfo>(){}.getType();
        apiClient.executeAsync(call, localVarReturnType, callback);
        return call;
    }
}
